buffer/src/main/java/io/netty/buffer/CompositeByteBuf.java

```diff
@@ -1836,7 +1836,10 @@ public CompositeByteBuf discardReadBytes() {
         }
 
         // Replace the first readable component with a new slice.
-        int trimmedBytes = readerIndex - c.offset;
+        int trimmedBytes = 0;
+        if (c != null) {
```

Comment:
This "fix" not looks valid at all... Also it would just throw an NPE under the if blog. Can you show me the NPE you are seeing ?